src/v1/internal/ch-websocket.js

```diff
@@ -233,4 +222,46 @@ function asWindowsFriendlyIPv6Address(scheme, parsedUrl) {
   return `${scheme}://${ipv6Host}:${parsedUrl.port}`;
 }
 
+/**
+ * @param {ChannelConfig} config - configuration for the channel.
+ * @param {function(): string} protocolSupplier - function that detects protocol of the web page.
+ * @return {{scheme: string|null, error: Neo4jError|null}} object containing either scheme or error.
+ */
+function determineWebSocketScheme(config, protocolSupplier) {
+  const encrypted = config.encrypted;
+  const trust = config.trust;
+
+  if (encrypted === false || encrypted === ENCRYPTION_OFF) {
+    // encryption explicitly turned off in the config
+    return {scheme: 'ws', error: null};
+  }
+
+  const protocol = typeof protocolSupplier === 'function' ? protocolSupplier() : '';
+  if (protocol && protocol.toLowerCase().indexOf('https') >= 0) {
+    // driver is used in a secure https web page, use 'wss'
+    return {scheme: 'wss', error: null};
+  }
+
+  if (encrypted === true || encrypted === ENCRYPTION_ON) {
+    // encryption explicitly requested in the config
+    if (!trust || trust === 'TRUST_CUSTOM_CA_SIGNED_CERTIFICATES') {
```

Comment:
This is totally out of scope of this PR, but I find `TRUST_CUSTOM_CA_SIGNED_CERTIFICATES` setting to be confusing and would prefer `TRUST_SYSTEM_CA_SIGNED_CERTIFICATES` as the default. We may consider revising naming / behaviour in upcoming releases?

Reply:
I'll create a card for this